File: src/main/java/uk/co/lukestevens/config/ApplicationProperties.java
package uk.co.lukestevens.config;

/**
 * An interface for retrieving properties
 * about the application itself like name and version.
 * 
 * @author Luke Stevens
 */
public interface ApplicationProperties {
	
	/**
	 * @return The version of this application
	 */
	String getApplicationVersion();
	
	/**
	 * @return The name of this application
	 */
	String getApplicationName();

	/**
	 * @return The group id of this application
	 */
	String getApplicationGroup();
}

File: src/main/kotlin/uk/co/lukestevens/config/ApplicationProperties.kt
package uk.co.lukestevens.config

/**
 * An interface for retrieving properties
 * about the application itself like name and version.
 *
 * @author Luke Stevens
 */
interface ApplicationProperties {
	/**
	 * @return The version of this application
	 */
	val applicationVersion: String

	/**
	 * @return The name of this application
	 */
	val applicationName: String

	/**
	 * @return The group id of this application
	 */
	val applicationGroup: String
}